### PR TITLE
add preview and writeFile options

### PR DESCRIPTION
**Pull request by jhford**
This is to aid in local debugging as well as to make it possible to process a JSON schema file on the command line

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,11 +17,12 @@
   "dependencies": {
     "ajv": "^4.0.4",
     "app-root-dir": "^1.0.2",
-    "aws-sdk": "^2.2.38",
+    "aws-sdk": "^2.5.3",
```
Comment (jhford) [inline package.json:20]: not sure how this crept in, but i don't see it being a problem...

**Comment by jonasfj** [comment]
I can't see the danger of adding this... I still think it would be more appropriate to add this functionality in tc-lib-docs...

I quickly looked at tc-lib-docs, but it has a long list of problems... such as using streams without being streaming, and not supporting retries, we probably need to spend a little more time on tc-lib-docs..

@imbstack what do you think? Wouldn't it be easier to do this through tc-lib-docs?

**Review by jonasfj — COMMENTED** [review]
I would like @imbstack thoughts...

I don't oppose the idea, just thinking it's better added in tc-lib-docs, when we want to remove publishing from tc-lib-validate/api/pulse-publisher

**Comment by jhford** [comment]
This is nothing to do with docs, this is a debugging tool for working on something which uses lib-validate.  Since lib-validate and thus taskcluster use non-standard json-schema formats, it's really nice to have the lib-validate library be self-standing for anyone who wishes to interoperate with taskcluster without having to use lib-docs to do basic debugging.

**Comment by jonasfj** [comment]
Fair point... And yeah, I don't really see an danger in adding it to this library.

**Review by imbstack — APPROVED** [review]

**Pull request merged**